### ISSUE TITLE
Split requirements into production and dev requirements

### ISSUE THREAD

**Issue by sebastianneubauer** [altair-viz/altair#768]
currently there are typical dev requirements (e.g. pytest, docutils and ipython) in the install_requires. That means one cannot install altair without installing those development requirements which is suboptimal or even sometimes not possible in production environments.  

A best practice would be to split the requirements into the minimal set of needed requirements and additional development extra requirements.

**Comment by sebastianneubauer** [comment]
I created a PR #769 which introduced a set of extra requirements for development environments. In production environments, e.g. in a backend server one can install via:
`pip install altair` 
In a development environment one can install additionally the dev requirements with:
`pip install -e .[dev]`

There is one issue remaining: To me, ipython should be a development requirement, but currently as it seems, altair is failing without ipython installed. So my suggestion would be to make ipython imports optional, e.g. something like this 
```
try:
    import ipython
except ImportError:
    NO_IPYTHON=True
```
What are your opinions?

**Comment by jakevdp** [comment]
I think making the IPython import optional is fine.

The only place where that would be nontrivial is in ``altair/_magics``, because the decorators are called when the submodule is imported.

That said, I don't see any reason that ``altair._magics`` would need to be imported except when IPython is available, so we're probably fine not changing anything there.
The other places that IPython is imported, a ``try``/``except`` should be fine.

**Comment by sebastianneubauer** [comment]
resolved by #769
